flocker/node/agents/test/test_blockdevice.py

```diff
@@ -10,18 +10,401 @@
+
 from ..blockdevice import (
-    LoopbackBlockDeviceAPI, IBlockDeviceAPI,
+    BlockDeviceDeployer, LoopbackBlockDeviceAPI, IBlockDeviceAPI,
     BlockDeviceVolume, UnknownVolume, AlreadyAttachedVolume,
-    _losetup_list_parse, _losetup_list, UnattachedVolume
+    CreateBlockDeviceDataset, UnattachedVolume,
+    _losetup_list_parse, _losetup_list,
 )
 
-from twisted.python.filepath import FilePath
-from twisted.trial.unittest import SynchronousTestCase, SkipTest
+from ... import InParallel, IDeployer, IStateChange
+from ....control import Dataset, Manifestation, Node, NodeState, Deployment
 
-GIBIBYTE = 2**30
+GIBIBYTE = 2 ** 30
 REALISTIC_BLOCKDEVICE_SIZE = 4 * GIBIBYTE
 
+if not platform.isLinux():
+    # The majority of Flocker isn't supported except on Linux - this test
+    # module just happens to run some code that obviously breaks on some other
+    # platforms.  Rather than skipping each test module individually it would
+    # be nice to have some single global solution.  FLOC-1560, FLOC-1205
+    skip = "flocker.node.agents.blockdevice is only supported on Linux"
+
+
+class BlockDeviceDeployerTests(SynchronousTestCase):
+    """
+    Tests for ``BlockDeviceDeployer``.
+    """
+    def test_interface(self):
+        """
+        ``BlockDeviceDeployer`` instances provide ``IDeployer``.
+        """
+        api = LoopbackBlockDeviceAPI.from_path(self.mktemp())
+        self.assertTrue(
+            verifyObject(
+                IDeployer,
+                BlockDeviceDeployer(
+                    hostname=u'192.0.2.123',
+                    block_device_api=api
+                )
+            )
+        )
+
+
+class BlockDeviceDeployerDiscoverLocalStateTests(SynchronousTestCase):
+    """
+    Tests for ``BlockDeviceDeployer.discover_local_state``.
+    """
+    def setUp(self):
+        self.expected_hostname = u'192.0.2.123'
+        self.api = loopbackblockdeviceapi_for_test(self)
+        self.deployer = BlockDeviceDeployer(
+            hostname=self.expected_hostname,
+            block_device_api=self.api
+        )
+
+    def assertDiscoveredState(self, deployer, expected_manifestations):
+        """
+        Assert that the manifestations on the state object returned by
+        ``deployer.discover_local_state`` equals the given list of
+        manifestations.
+
+        :param IDeployer deployer: The object to use to discover the state.
+        :param list expected_manifestations: The ``Manifestation``\ s expected
+            to be discovered.
+
+        :raise: A test failure exception if the manifestations are not what is
+            expected.
+        """
+        discovering = deployer.discover_local_state()
+        state = self.successResultOf(discovering)
+        expected_paths = {}
+        for manifestation in expected_manifestations:
+            dataset_id = manifestation.dataset.dataset_id
+            mountpath = deployer._mountpath_for_manifestation(manifestation)
```

Comment:
This class claims to be for `BlockDeviceDeployer` generically, but here it's using a private method that is loopback specific (`_mountpath_for_manifestation`).

I don't think asserting *specific* values for paths is important (it's not part of public API), you should merely assert that the paths are present.